factory/post.js

```diff
@@ -26,7 +25,10 @@ function processContent(data) {
     var item = entries[i],
         postTitle = item.title.$t,
         content   = item.content.$t,
-        url       = item.link[4].href,
+        // Do not assume the link will always remain item 4 in the array
+        url       = _.filter(item.link, function (link) {
+          return link.rel === 'alternate' && link.type === 'text/html';
+        })[0].href,
```

Comment:
Very detailed explanation. Just one thought tho, what if Google decides to add a new link like you said earlier, a new alternate link of type text/html too. What happens in that scenario 😄  ?